Lib/inspect.py

```diff
@@ -1065,15 +1065,21 @@ def __init__(self):
 
     def tokeneater(self, type, token, srowcol, erowcol, line):
         if not self.started and not self.indecorator:
-            # skip any decorators
-            if token == "@":
-                self.indecorator = True
-            # look for the first "def", "class" or "lambda"
-            elif token in ("def", "class", "lambda"):
-                if token == "lambda":
+            if type != tokenize.INDENT:
+                # skip any decorators
+                if token == "@":
+                    self.indecorator = True
+                elif token == "async":
+                    pass
+                # look for the first "def", "class" or "lambda"
+                elif token in ("def", "class", "lambda"):
+                    if token == "lambda":
+                        self.islambda = True
+                    self.started = True
+                else:
                     self.islambda = True
```

Comment:
Could you explain this part a bit to me? What's in the else case? Why is `islambda` set? I was a bit confused (without testing too much into it) when I read the code because I thought this is definitely not a lambda here.